src/main/java/bio/terra/controller/GlobalExceptionHandler.java

```diff
@@ -46,15 +47,25 @@ public ErrorModel validationExceptionHandler(Exception ex) {
         return buildErrorModel(ex);
     }
 
+    // -- job response exception -- we use the JobResponseException to wrap non-runtime exceptions
+    // returned from flights. So at this level, we catch the JobResponseException and retrieve the
+    // original exception from inside it and use that to construct the error model.
+    @ExceptionHandler(JobResponseException.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public ErrorModel jobResponseExceptionHandler(Exception ex) {
+        Throwable nestedException = ex.getCause();
+        return buildErrorModel(nestedException);
+    }
+
     // -- catchall - log so we can understand what we have missed in the handlers above
-    @ExceptionHandler(Exception.class)
+    @ExceptionHandler(Throwable.class)
     @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
     public ErrorModel catchallHandler(Exception ex) {
```

Comment:
I think this type now needs to be Throwable. 
I also noticed in StepResult that Throwable has been changed to Exception. I'm not sure which we really want, but I think we should be consistent here and in StepResult. 
My understanding is that Throwable are things that the application typically can not handle, like out of memory. Not sure if those are things we should be returning to the client.

Reply:
Right. I think the only thing we should deal with in our code is Exception. We never catch any non-exception. I don't think this code needs to deal with Throwables.

The only reason to have Throwable in the buildErrorModel is because the type of getCause() in the JobResponseException handler is Throwable.

If it reads better, I can make buildErrorModel take Exception and duplicate the code inside of the special case of JobResponseException.

Reply:
ah, maybe it is better to go back to the code in the original PR. and maybe add a comment for buildErrorModel that says it takes a Throwable because that is the type of the contained cause

Reply:
OK, now I use Exception everywhere except that one method; and I commented that method.
Thanks for your help!